src/users/users.service.ts

```diff
@@ -10,38 +10,91 @@ import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class UsersService {
-  constructor(private prisma: PrismaService) {}
+  constructor(private prisma: PrismaService) { }
 
-  private sanitizeOutput(user: User): SanitizedUser {
+  public sanitizeOutput(user: User): SanitizedUser {
     return exclude(user, ['id', 'emailNonce', 'passwordNonce', 'password', 'createdAt', 'updatedAt']);
   }
 
   private async passwd_encrypt(password: string): Promise<string> {
     return await bcrypt.hash(password, 10);
   }
 
-  async create(user: CreateUserDto): Promise<SanitizedUser> {
-    return this.sanitizeOutput(await this.prisma.user.create({
+  async create(user: CreateUserDto): Promise<User> {
+    return await this.prisma.user.create({
       data: {
         username: user.username,
         email: user.email,
         password: await this.passwd_encrypt(user.password),
-        isAdmin: false,
+        isAdmin: false
       }
-    }));
+    });
   }
 
   async validateEmail(email: string): Promise<boolean> {
-      // Check that email address doesn't already exist in the db
-      return !await this.findOne({email});
+    // Check that email address doesn't already exist in the db
+    return !await this.findOne({ email });
+  }
+
+  async validateEmailNonce(email_nonce: string): Promise<boolean> {
+    try {
+      await this.prisma.user.update({
+        where: {
+          emailNonce: email_nonce,
+        },
+        data: {
+          emailNonce: null
+        }
+      })
+      return true;
+    } catch (err) {
+      return false;
+    }
+
+  }
+
+
+  async updatePassword(id: string, password: string): Promise<boolean> {
+    try {
+      await this.prisma.user.update({
+        where: {
+          id: id
+        },
+        data: {
+          passwordNonce: null,
+          password: await this.passwd_encrypt(password)
+        }
+      })
+      return true;
+    } catch (err) {
+      return false;
+    }
+
   }
 
+  async updatePasswordNonce(token: string, id: string): Promise<boolean> {
+    try {
+      await this.prisma.user.update({
+        where: {
+          id: id,
+        },
+        data: {
+          passwordNonce: token
+        }
+      })
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
```

Comment:
I recommend not accepting strings from outside for the regenerated token.

```suggestion
  async regeneratePasswordNonce(id: string): Promise<boolean> {
    try {
      await this.prisma.user.update({
        where: {
          id: id,
        },
        data: {
          passwordNonce: uuidv4()
        }
      })
      return true;
    } catch (err) {
      return false;
    }
  }
```